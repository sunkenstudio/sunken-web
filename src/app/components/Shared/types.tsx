export interface StrapiImage {
  typename: string;
  url: string;
}

export interface StrapiStyledImage {
  typename: string;
  media: StrapiImage;
  alt: string;
  borderWidth: string;
  borderColor: string;
  borderRadius: string;
  grayscalePercent: number;
  filterColor: string;
  filterOpacity: number;
}

export interface RichTextParagraph {
  type: string;
  children: RichTextSegment[];
}

type RichTextSegment = {
  type: "text" | "link";
  text?: string;
  url?: string;
  bold?: boolean;
  italic?: boolean;
  children?: RichTextSegment[];
};

export interface StrapiSection {
  typename: string;
  header: string;
  sortOrder: number;
  text: RichTextParagraph[];
  image: StrapiImage;
  caption: string;
  bgImageOpacity: number;
  bgFilterOpacity: number;
  bgImage: StrapiImage;
}

export interface StrapiBrand {
  typename: string;
  primary: string;
  secondary: string;
  accent: string;
  light: string;
  dark: string;
  font: {
    family: string;
    url: string;
  };
}

export interface StrapiFooter {
  typename: string;
  text: RichTextParagraph[];
  facebookUrl: string;
  instagramUrl: string;
  twitterUrl: string;
}

export interface StrapiHero {
  typename: string;
  header: string;
  subheader: string;
  button1: string;
  button2: string;
  text: RichTextParagraph[];
  image: StrapiStyledImage;
  bgImageOpacity: number;
  bgFilterOpacity: number;
  bgImage: StrapiImage;
}

export interface StrapiContact {
  typename: string;
  bgImage: StrapiImage;
  bgImageOpacity: number;
  bgFilterOpacity: number;
}

export interface Client {
  typename: string;
  clientId: string;
  hero: StrapiHero;
  sections: StrapiSection[];
  brand: StrapiBrand;
  footer: StrapiFooter;
  contact: StrapiContact;
}
