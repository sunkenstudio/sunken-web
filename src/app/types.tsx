import { type BlocksContent } from '@strapi/blocks-react-renderer';

export interface StrapiContact {
  typename: string;
  fields: StrapiInputField[];
  sendTo: string;
  header: string;
  bgColor: Color;
  bgImage: StrapiStyledImage;
}

export interface StrapiInputField {
  label: string;
  type:
    | 'text'
    | 'email'
    | 'tel'
    | 'calendar'
    | 'time'
    | 'time_range'
    | 'dropdown'
    | 'money'
    | 'number'
    | 'textarea'
    | 'date';
  options?: string[];
}

export interface StrapiCoordinates {
  lat: number;
  lng: number;
  label?: string;
}

export interface StrapiConfig {
  typename: string;
  isUnderConstruction: boolean;
  isMaintenanceMode: boolean;
}

export interface StrapiImage {
  typename: string;
  url: string;
}

export type Color = 'primary' | 'secondary' | 'accent' | 'light' | 'dark';

export type StrapiBrandColor = {
  color: Color;
} | null;

export interface StrapiStyledImage {
  typename: string;
  media: StrapiImage;
  alt: string;
  border: StrapiBorder;
  filter: StrapiFilter;
}

export type StrapiBorder = {
  width: string | null;
  color: Color;
  radius: string | null;
} | null;

type StrapiFilter = {
  color: Color;
  opacity: number;
  grayscale: number;
} | null;

export interface StrapiStyledButton {
  typename: string;
  type: 'link' | 'modal' | 'pdf' | 'submit';
  text: string;
  href: string;
  icon: string | null;
  color: string;
  bgColor: string;
  border: StrapiBorder;
  shadow: string;
}

export interface StrapiSection {
  typename: string;
  header: string;
  variant: 'left' | 'right' | 'splitLeft' | 'splitRight';
  sortOrder: number;
  text: BlocksContent;
  image: StrapiStyledImage;
  caption: string | null;
  bgImage: StrapiStyledImage | null;
  bgColor: Color;
}

export interface StrapiBrand {
  typename: string;
  primary: string;
  secondary: string;
  accent: string;
  light: string;
  dark: string;
  fonts: {
    typename: string;
    family: string;
    url: string;
  }[];
  companyName: string;
}

export interface StrapiFooter {
  typename: string;
  text: BlocksContent;
  facebookUrl: string;
  instagramUrl: string;
  twitterUrl: string;
}

export interface StrapiHero {
  typename: string;
  header: string;
  variant: 'leftAligned' | 'centerAligned' | 'split';
  subheader: string;
  buttons: StrapiStyledButton[];
  text: BlocksContent;
  image: StrapiStyledImage | null;
  bgImageOpacity: number;
  bgFilterOpacity: number;
  bgImage: StrapiStyledImage;
}

export interface StrapiMapSection {
  typename: string;
  theme: string;
  center: StrapiCoordinates;
  markers: StrapiCoordinates[];
  markerIcon: string;
  zoom: number;
}

export interface Client {
  typename: string;
  clientId: string;
  hero: StrapiHero;
  sections: StrapiSection[];
  brand: StrapiBrand;
  footer: StrapiFooter;
  contact: StrapiContact;
  config: StrapiConfig;
  mapSection: StrapiMapSection;
}
