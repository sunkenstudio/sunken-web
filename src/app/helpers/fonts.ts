import { StrapiBrand } from "../components/Shared/types";

const FontFaceObserver = require("fontfaceobserver");

const Fonts = (font: StrapiBrand["font"]) => {
  const { url, family } = font;
  const link = document.createElement("link");
  link.href = url;
  link.rel = "stylesheet";

  document.head.appendChild(link);

  const roboto = new FontFaceObserver(family);

  roboto.load().then(() => {
    document.documentElement.classList.add(family.toLowerCase());
  });
};

export default Fonts;
