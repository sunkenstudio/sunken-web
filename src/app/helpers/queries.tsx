import { gql } from "@apollo/client";

export const GET_SITE = gql`
  query Site($ClientId: String) {
    sites(filters: { ClientId: { eq: $ClientId } }) {
      data {
        attributes {
          ClientId
          hero {
            data {
              attributes {
                Header
                Subheader
                Button1
                Button2
                Text
                Image {
                  data {
                    attributes {
                      url
                    }
                  }
                }
                BgImageOpacity
                BgFilterOpacity
                BgImage {
                  data {
                    attributes {
                      url
                    }
                  }
                }
              }
            }
          }
          sections {
            data {
              attributes {
                Header
                SortOrder
                Text
                Image {
                  data {
                    attributes {
                      url
                    }
                  }
                }
                Caption
                BgImageOpacity
                BgFilterOpacity
                BgImage {
                  data {
                    attributes {
                      url
                    }
                  }
                }
              }
            }
          }
          brand {
            data {
              attributes {
                Primary
                Secondary
                Accent
                Light
                Dark
                Font {
                  Url
                  Family
                }
              }
            }
          }
          footer {
            data {
              attributes {
                Text
                FacebookUrl
                InstagramUrl
                TwitterUrl
              }
            }
          }
          contact {
            data {
              attributes {
                BgImage {
                  data {
                    attributes {
                      url
                    }
                  }
                }
                BgImageOpacity
                BgFilterOpacity
              }
            }
          }
        }
      }
    }
  }
`;
