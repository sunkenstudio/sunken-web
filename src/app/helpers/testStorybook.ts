// @ts-nocheck
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/react';
import preview from '../../../.storybook/preview';

export const testStories = (Stories: any) => {
  const wrapper = preview?.decorators?.[0];
  const testCases = Object.entries(composeStories(Stories));

  test.each(testCases)('Renders %s story', async (_StoryName, Story) => {
    const tree = await render(wrapper(Story, Story));
    expect(tree.baseElement).toMatchSnapshot();
  });
};
