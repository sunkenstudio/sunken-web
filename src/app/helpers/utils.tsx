'use client';
import { camelCase } from 'lodash';
import { Client } from '../types';

export const emptyFunction = () => {
  // Default value for no behavior
};

export const scrollToElement = (e: any) => {
  e.preventDefault();

  const href = e.currentTarget.href;
  const targetId = href.replace(/.*#/, '');

  const elem = document.getElementById(targetId);
  elem?.scrollIntoView({
    behavior: 'smooth',
  });
};

export const getClientIdFromUrl = () => {
  // localhost
  const url = window?.location?.href;
  if (url.includes('localhost')) {
    const urlParams = new URLSearchParams(window.location.search);
    const clientId = urlParams.get('clientId');
    return clientId;
  }
  // prod
  return window?.location?.hostname;
};

// TODO - Figure out the correct typing for this mess
export const formatStrapiData = (data: any): Client => {
  return Object.entries(data).reduce((acc, [key, val]) => {
    const formattedKey = camelCase(key);
    if (typeof val === 'object') {
      /* @ts-ignore */
      if (val && Array.isArray(val?.data)) {
        return {
          ...acc,
          /* @ts-ignore */
          [formattedKey]: val.data.map((i) => formatStrapiData(i.attributes)),
        };
      }
      /* @ts-ignore */
      if (val?.data?.attributes) {
        return {
          ...acc,
          /* @ts-ignore */
          [formattedKey]: formatStrapiData(val.data.attributes),
        };
      }
    }
    /* @ts-ignore */
    if (val?.__typename?.includes('Component')) {
      return {
        ...acc,
        [formattedKey]: formatStrapiData(val),
      };
    }
    // hero > buttons array
    if (
      val &&
      Array.isArray(val) &&
      val?.[0]?.__typename?.includes('ComponentCommon')
    ) {
      return {
        ...acc,
        [formattedKey]: val.map((i) => formatStrapiData(i)),
      };
    }
    return {
      ...acc,
      [formattedKey]: val,
    };
  }, {}) as unknown as Client;
};
