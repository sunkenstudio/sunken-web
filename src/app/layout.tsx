import './globals.css';
import { Providers } from './providers';

export const metadata = {
  // title: "Sunken Studio",
  // description: "Generated by create next app",
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <body>
        <Providers>{children}</Providers>
      </body>
    </html>
  );
}
