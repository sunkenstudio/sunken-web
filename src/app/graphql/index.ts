import { gql } from '@apollo/client';
import * as Queries from './queries';

export const GET_SITE_ID = gql`
  query GetSiteIdByClientId($ClientId: String) {
    sites(filters: { ClientId: { eq: $ClientId } }, pagination: { limit: 1 }) {
      data {
        id
      }
    }
  }
`;
export const GET_SITE = gql`
  ${Queries.GET_HERO}
  ${Queries.SECTIONS}
  ${Queries.PROJECT_SECTION}
  ${Queries.BRAND}
  ${Queries.CAROUSEL}
  ${Queries.FOOTER}
  ${Queries.CONTACT}
  ${Queries.CONFIG}
  ${Queries.IMAGE_SCROLL}
  ${Queries.FEATURE_SECTION}

  query Site($id: ID!) {
    site(id: $id) {
      data {
        attributes {
          ClientId
          hero {
            ...hero
          }
          sections {
            ...sections
          }
          project_section {
            ...projectSection
          }
          brand {
            ...brand
          }
          carousel {
            ...carousel
          }
          footer {
            ...footer
          }
          contact {
            ...contact
          }
          config {
            ...config
          }
          image_scroll {
            ...imageScroll
          }
          feature_section {
            ...featureSection
          }
        }
      }
    }
  }
`;

export const GET_IMAGES = gql`
  query {
    uploadFiles {
      data {
        id
        attributes {
          name
          url
          alternativeText
          caption
        }
      }
    }
  }
`;
