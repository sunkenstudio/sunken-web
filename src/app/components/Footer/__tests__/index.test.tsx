'use client';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

import { Footer, FooterProps } from '..';
import { FooterFixture } from '@/app/fixtures';
import { Color } from '@/app/types';
import { testStories } from '@/app/helpers/testStorybook';
import * as Stories from '../index.stories';

describe('Footer', () => {
  const renderComponent = (overrides: Partial<FooterProps> = {}) => {
    const defaults = (props: Partial<FooterProps>): FooterProps => ({
      buttonStyle: {
        textColor: 'black',
        bgColor: 'white',
        border: {
          width: '0px',
          color: 'dark' as Color,
          radius: '0px',
        },
        shadow: '',
      },
      footer: FooterFixture(),
      ...props,
    });
    return render(<Footer {...defaults(overrides)} />);
  };

  describe('All stories should render successfully', () => {
    testStories(Stories);
  });

  it('match snapshot', () => {
    const rendered = renderComponent();
    expect(rendered).toMatchSnapshot();
  });

  describe('social icons', () => {
    it('if given facebook url, should render facebook icon', () => {
      const rendered = renderComponent({
        footer: FooterFixture({ facebookUrl: 'http://facebook.com' }),
      });
      const icon = rendered.getByTitle('Facebook Logo');
      expect(icon).toBeDefined();
    });

    it('if NOT given facebook url, should NOT render facebook icon', () => {
      const rendered = renderComponent({
        footer: FooterFixture({ facebookUrl: '' }),
      });
      const icon = rendered.queryByTitle('Facebook Logo');
      expect(icon).toBeNull();
    });

    it('if given twitter url, should render twitter icon', () => {
      const rendered = renderComponent({
        footer: FooterFixture({ twitterUrl: 'http://twitter.com' }),
      });
      const icon = rendered.getByTitle('Twitter Logo');
      expect(icon).toBeDefined();
    });

    it('if NOT given twitter url, should NOT render twitter icon', () => {
      const rendered = renderComponent({
        footer: FooterFixture({ twitterUrl: '' }),
      });
      const icon = rendered.queryByTitle('Twitter Logo');
      expect(icon).toBeNull();
    });

    it('if given instagram url, should render instagram icon', () => {
      const rendered = renderComponent({
        footer: FooterFixture({ instagramUrl: 'http://instagram.com' }),
      });
      const icon = rendered.getByTitle('Instagram Logo');
      expect(icon).toBeDefined();
    });

    it('if NOT given instagram url, should NOT render instagram icon', () => {
      const rendered = renderComponent({
        footer: FooterFixture({ instagramUrl: '' }),
      });
      const icon = rendered.queryByTitle('Instagram Logo');
      expect(icon).toBeNull();
    });
  });
});
