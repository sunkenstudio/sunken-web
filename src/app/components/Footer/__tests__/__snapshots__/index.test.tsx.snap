// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-do35vm"
      >
        <div
          class="chakra-stack css-1wrik5w"
        >
          <div
            class="chakra-stack css-j8nvvj"
          >
            <a
              href="https://instagram.com"
              target="_blank"
            >
              <a
                class="chakra-button css-1fvmww3"
              >
                <svg
                  fill="currentColor"
                  height="32"
                  viewBox="0 0 256 256"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title>
                    Instagram Logo
                  </title>
                  <path
                    d="M128,80a48,48,0,1,0,48,48A48.05,48.05,0,0,0,128,80Zm0,80a32,32,0,1,1,32-32A32,32,0,0,1,128,160ZM176,24H80A56.06,56.06,0,0,0,24,80v96a56.06,56.06,0,0,0,56,56h96a56.06,56.06,0,0,0,56-56V80A56.06,56.06,0,0,0,176,24Zm40,152a40,40,0,0,1-40,40H80a40,40,0,0,1-40-40V80A40,40,0,0,1,80,40h96a40,40,0,0,1,40,40ZM192,76a12,12,0,1,1-12-12A12,12,0,0,1,192,76Z"
                  />
                </svg>
              </a>
            </a>
            <a
              href="https://facebook.com"
              target="_blank"
            >
              <a
                class="chakra-button css-1fvmww3"
              >
                <svg
                  fill="currentColor"
                  height="32"
                  viewBox="0 0 256 256"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title>
                    Facebook Logo
                  </title>
                  <path
                    d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm8,191.63V152h24a8,8,0,0,0,0-16H136V112a16,16,0,0,1,16-16h16a8,8,0,0,0,0-16H152a32,32,0,0,0-32,32v24H96a8,8,0,0,0,0,16h24v63.63a88,88,0,1,1,16,0Z"
                  />
                </svg>
              </a>
            </a>
            <a
              href="https://twitter.com"
              target="_blank"
            >
              <a
                class="chakra-button css-1fvmww3"
              >
                <svg
                  fill="currentColor"
                  height="32"
                  viewBox="0 0 256 256"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title>
                    Twitter Logo
                  </title>
                  <path
                    d="M247.39,68.94A8,8,0,0,0,240,64H209.57A48.66,48.66,0,0,0,168.1,40a46.91,46.91,0,0,0-33.75,13.7A47.9,47.9,0,0,0,120,88v6.09C79.74,83.47,46.81,50.72,46.46,50.37a8,8,0,0,0-13.65,4.92c-4.31,47.79,9.57,79.77,22,98.18a110.93,110.93,0,0,0,21.88,24.2c-15.23,17.53-39.21,26.74-39.47,26.84a8,8,0,0,0-3.85,11.93c.75,1.12,3.75,5.05,11.08,8.72C53.51,229.7,65.48,232,80,232c70.67,0,129.72-54.42,135.75-124.44l29.91-29.9A8,8,0,0,0,247.39,68.94Zm-45,29.41a8,8,0,0,0-2.32,5.14C196,166.58,143.28,216,80,216c-10.56,0-18-1.4-23.22-3.08,11.51-6.25,27.56-17,37.88-32.48A8,8,0,0,0,92,169.08c-.47-.27-43.91-26.34-44-96,16,13,45.25,33.17,78.67,38.79A8,8,0,0,0,136,104V88a32,32,0,0,1,9.6-22.92A30.94,30.94,0,0,1,167.9,56c12.66.16,24.49,7.88,29.44,19.21A8,8,0,0,0,204.67,80h16Z"
                  />
                </svg>
              </a>
            </a>
          </div>
          <span>
            Made with &lt;3 in Detroit, MI
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-do35vm"
    >
      <div
        class="chakra-stack css-1wrik5w"
      >
        <div
          class="chakra-stack css-j8nvvj"
        >
          <a
            href="https://instagram.com"
            target="_blank"
          >
            <a
              class="chakra-button css-1fvmww3"
            >
              <svg
                fill="currentColor"
                height="32"
                viewBox="0 0 256 256"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Instagram Logo
                </title>
                <path
                  d="M128,80a48,48,0,1,0,48,48A48.05,48.05,0,0,0,128,80Zm0,80a32,32,0,1,1,32-32A32,32,0,0,1,128,160ZM176,24H80A56.06,56.06,0,0,0,24,80v96a56.06,56.06,0,0,0,56,56h96a56.06,56.06,0,0,0,56-56V80A56.06,56.06,0,0,0,176,24Zm40,152a40,40,0,0,1-40,40H80a40,40,0,0,1-40-40V80A40,40,0,0,1,80,40h96a40,40,0,0,1,40,40ZM192,76a12,12,0,1,1-12-12A12,12,0,0,1,192,76Z"
                />
              </svg>
            </a>
          </a>
          <a
            href="https://facebook.com"
            target="_blank"
          >
            <a
              class="chakra-button css-1fvmww3"
            >
              <svg
                fill="currentColor"
                height="32"
                viewBox="0 0 256 256"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Facebook Logo
                </title>
                <path
                  d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm8,191.63V152h24a8,8,0,0,0,0-16H136V112a16,16,0,0,1,16-16h16a8,8,0,0,0,0-16H152a32,32,0,0,0-32,32v24H96a8,8,0,0,0,0,16h24v63.63a88,88,0,1,1,16,0Z"
                />
              </svg>
            </a>
          </a>
          <a
            href="https://twitter.com"
            target="_blank"
          >
            <a
              class="chakra-button css-1fvmww3"
            >
              <svg
                fill="currentColor"
                height="32"
                viewBox="0 0 256 256"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Twitter Logo
                </title>
                <path
                  d="M247.39,68.94A8,8,0,0,0,240,64H209.57A48.66,48.66,0,0,0,168.1,40a46.91,46.91,0,0,0-33.75,13.7A47.9,47.9,0,0,0,120,88v6.09C79.74,83.47,46.81,50.72,46.46,50.37a8,8,0,0,0-13.65,4.92c-4.31,47.79,9.57,79.77,22,98.18a110.93,110.93,0,0,0,21.88,24.2c-15.23,17.53-39.21,26.74-39.47,26.84a8,8,0,0,0-3.85,11.93c.75,1.12,3.75,5.05,11.08,8.72C53.51,229.7,65.48,232,80,232c70.67,0,129.72-54.42,135.75-124.44l29.91-29.9A8,8,0,0,0,247.39,68.94Zm-45,29.41a8,8,0,0,0-2.32,5.14C196,166.58,143.28,216,80,216c-10.56,0-18-1.4-23.22-3.08,11.51-6.25,27.56-17,37.88-32.48A8,8,0,0,0,92,169.08c-.47-.27-43.91-26.34-44-96,16,13,45.25,33.17,78.67,38.79A8,8,0,0,0,136,104V88a32,32,0,0,1,9.6-22.92A30.94,30.94,0,0,1,167.9,56c12.66.16,24.49,7.88,29.44,19.21A8,8,0,0,0,204.67,80h16Z"
                />
              </svg>
            </a>
          </a>
        </div>
        <span>
          Made with &lt;3 in Detroit, MI
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
